### DISCUSSION TITLE
What should I put into the private and public keys under the Service Worker settings?

### DISCUSSION THREAD

**Discussion by reinhart1010** [misskey-dev/misskey#9196] [Q&A] [answered]
First thing first, I know that I should input a key pair into the "Service Worker" settings.

<img width="510" alt="image" src="https://user-images.githubusercontent.com/17312341/202705682-27fa89fa-2368-48d8-9778-b232523ef4f8.png">

But in what specific format? Like the OpenSSH, ASCII-armored PGP, or what else?

**Comment by Trif4** [comment, marked as answer]
They're [VAPID keys](https://web.dev/push-notifications-web-push-protocol/). You can generate a key pair with

`npx web-push generate-vapid-keys`

**Reply by ThatOneCalculator** [reply]
I feel like Misskey should explain this better in the docs, I added it to my fork's readme here: https://codeberg.org/thatonecalculator/calckey/#tips-tricks

**Comment by prof-xadk3** [comment]
Functional AES wrapper around wss streams might also help. 😁
